gcode/src/parser.rs

```diff
@@ -404,36 +414,54 @@ mod tests {
         assert_eq!(got[1].gcodes().len(), 1);
     }
 
-    /// I wasn't sure if the `#[derive(Serialize)]` would work given we use
-    /// `B::Comments`, which would borrow from the original source.
     #[test]
-    #[cfg(feature = "serde-1")]
-    fn you_can_actually_serialize_lines() {
-        let src = "G01 X5 G90 (comment) G91 M10\nG01\n";
-        let line = parse(src).next().unwrap();
+    fn funny_bug_in_crate_example() {
+        let src = "G90 \n G01 X50.0 Y-10";
+        let expected = vec![
+            GCode::new(Mnemonic::General, 90.0, Span::PLACEHOLDER),
+            GCode::new(Mnemonic::General, 1.0, Span::PLACEHOLDER)
+                .with_argument(Word::new('X', 50.0, Span::PLACEHOLDER))
+                .with_argument(Word::new('Y', -10.0, Span::PLACEHOLDER)),
+        ];
+
+        let got: Vec<_> = crate::parse(src).collect();
+        assert_eq!(got, expected);
+    }
 
-        fn assert_serializable<S: serde::Serialize>(_: &S) {}
-        fn assert_deserializable<'de, D: serde::Deserialize<'de>>() {}
+    #[test]
+    #[ignore]
+    fn implicit_command_after_newline() {
+        let src = "G01 X1.0 Y2.0\n X3.0 Y4.0";
+        let expected = vec![
+            GCode::new(Mnemonic::General, 1.0, Span::PLACEHOLDER)
+                .with_argument(Word::new('X', 1.0, Span::PLACEHOLDER))
+                .with_argument(Word::new('Y', 2.0, Span::PLACEHOLDER)),
+            GCode::new(Mnemonic::General, 1.0, Span::PLACEHOLDER)
+                .with_argument(Word::new('X', 3.0, Span::PLACEHOLDER))
+                .with_argument(Word::new('Y', 4.0, Span::PLACEHOLDER)),
+        ];
 
-        assert_serializable(&line);
-        assert_deserializable::<Line<'_>>();
+        let got: Vec<_> = crate::parse(src).collect();
+        assert_eq!(got, expected);
     }
 
-    /// For some reason we were parsing the G90, then an empty G01 and the
-    /// actual G01.
     #[test]
     #[ignore]
-    fn funny_bug_in_crate_example() {
-        let src = "G90 \n G01 X50.0 Y-10";
+    // This test focuses on the G90 and M7 on the same line.
```

Comment:
If we are updating the test, should we also remove the `#[ignore]` attribute?

Reply:
Ah, this is just a dumb git diff. I did remove the `#[ignore]` for the test `funny_bug_in_crate_example()`, but I also added two more tests below it that are confusing it. If you look directly at the source it'll make sense.